### PR TITLE
replace boost log with spdlog

### PR DESCRIPTION
**Pull request by apppur**
## What is the purpose of the change

replace boost log with spdlog

## Brief changelog

replace boost log with spdlog

## Verifying this change

replace boost log with spdlog

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when a cross-module dependency exists.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: src/common/UtilAll.h [hunk 1]
```diff
@@ -82,9 +87,9 @@ inline void deleteAndZero(Type& pointer) {
 }
 #define EMPTY_STR_PTR(ptr) (ptr == NULL || ptr[0] == '\0')
 #ifdef WIN32
-#define SIZET_FMT "%lu"
+#define SIZET_FMT "{}"
```
[the remark this answers is not in the document]
Comment (apppur) [reply src/common/UtilAll.h]: sorry, I ignored the previous changes. now I revert it.

File: src/common/UtilAll.cpp [hunk 2]
```diff
@@ -267,6 +267,50 @@ string UtilAll::getHomeDirectory() {
...
+#else
+        int flag = mkdir(dir, 0755);
+#endif
+        if (flag == 0) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+	return true;
```
Comment (ifplusor) [inline src/common/UtilAll.cpp]: '\t' as indentation is bad. Use two whitespace please.

File: src/common/UtilAll.cpp [hunk 3]
```diff
@@ -267,6 +267,50 @@ string UtilAll::getHomeDirectory() {
...
+    if (dir == nullptr)  {
+        return false;
+    }
+    if (access(dir, 0) == -1) {
+#ifdef _WIN32
+        int flag = mkdir(dir);
+#else
+        int flag = mkdir(dir, 0755);
+#endif
+        if (flag == 0) {
```
Comment (ifplusor) [inline src/common/UtilAll.cpp]: I think use "return flag == 0;" directly is better, and 'ret' is better than 'flag'.

File: src/common/UtilAll.cpp [hunk 4]
```diff
@@ -267,6 +267,50 @@ string UtilAll::getHomeDirectory() {
...
+    }
+	return true;
+}
+
+void UtilAll::createDirectory(std::string const& dir) {
+    const char* ptr = dir.c_str();
+    if (access(ptr, 0) == 0) {
+        return;
+    }
+    char buff[PATH_MAX] = {0};
```
Comment (ifplusor) [inline src/common/UtilAll.cpp]: undefined PATH_MAX in windows, instead of MAX_PATH

**Comment by ifplusor** [comment]
Already done in the re_dev branch.

**Pull request closed without merging**